File: sidecars/synaplan-compute/tests/hostile/dns_attempt.py
# expected-result: succeeded
# reason:
# note: Name resolution fails; no packet leaves (host tcpdump in CI)
"""dns_attempt.py — try to resolve an external name; success of the control is failure of DNS."""
import socket
import sys

try:
    socket.setdefaulttimeout(2)
    socket.getaddrinfo("example.com", 443)
except OSError:
    sys.exit(0)
print("dns unexpectedly succeeded", file=sys.stderr)
sys.exit(2)

File: sidecars/synaplan-compute/tests/hostile/node/dns_attempt.js
# expected-result: succeeded
# reason:
# note: Name resolution fails; no packet leaves
"""dns_attempt.js — resolve example.com; control success is DNS failure."""
const dns = require('dns');
dns.setServers([]);
dns.lookup('example.com', { timeout: 2000 }, (err) => {
  if (err) {
    process.exit(0);
  }
  process.stderr.write('dns unexpectedly succeeded\n');
  process.exit(2);
});
